spec/facility_spec.rb

```diff
@@ -2,25 +2,275 @@
   end
+  
   describe '#initialize' do
     it 'can initialize' do
-      expect(@facility).to be_an_instance_of(Facility)
-      expect(@facility.name).to eq('DMV Tremont Branch')
-      expect(@facility.address).to eq('2855 Tremont Place Suite 118 Denver CO 80205')
-      expect(@facility.phone).to eq('(720) 865-4600')
-      expect(@facility.services).to eq([])
+      expect(@facility_1).to be_an_instance_of(Facility)
+      expect(@facility_1.name).to eq('DMV Tremont Branch')
+      expect(@facility_1.address).to eq('2855 Tremont Place Suite 118 Denver CO 80205')
+      expect(@facility_1.phone).to eq('(720) 865-4600')
+      expect(@facility_1.services).to eq([])
+      expect(@facility_1.registered_vehicles).to eq([])
     end
   end
 
   describe '#add service' do
     it 'can add available services' do
-      expect(@facility.services).to eq([])
-      @facility.add_service('New Drivers License')
-      @facility.add_service('Renew Drivers License')
-      @facility.add_service('Vehicle Registration')
-      expect(@facility.services).to eq(['New Drivers License', 'Renew Drivers License', 'Vehicle Registration'])
+      expect(@facility_1.services).to eq([])
+      
+      @facility_1.add_service('New Drivers License')
+      @facility_1.add_service('Renew Drivers License')
+      @facility_1.add_service('Vehicle Registration')
+      
+      expect(@facility_1.services).to eq(['New Drivers License', 'Renew Drivers License', 'Vehicle Registration'])
     end
   end
+
+  describe '#register_vehicle(vehicle)' do 
+    it 'has no registered vehicles' do 
+      expect(@facility_1.registered_vehicles).to eq([])
+    end 
+    
+    it 'facility provides the service needed' do 
+      expect(@facility_1.services).to eq([])
+
+      @facility_1.add_service('Vehicle Registration')
+
+      expect(@facility_1.services).to eq(['Vehicle Registration'])
+    end 
+
+    it 'does not provide service if the facility does not offer it' do
+      expect(@facility_1.services).to eq([])
+
+      @facility_1.add_service('Vehicle Registration')
+
+      expect(@facility_2.services).to eq([])
+    end 
+
+    it 'can tell if a vehicle was registered' do 
+      cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice} )
+      
+      @facility_1.add_service('Vehicle Registration')
+      @facility_1.register_vehicle(cruz)
+
+      expect(@facility_1.registered_vehicles).to eq([cruz])
+    end 
+  
+    it 'can give a date a vehicle is registered' do
+      cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice} )
+
+            
+      @facility_1.add_service('Vehicle Registration')
+      @facility_1.register_vehicle(cruz)
+
+      expect(@facility_1.registered_vehicles.first.registration_date).to eq(Date.today)
+    end
```

Comment:
Great organization to this describe block for the register_vehicle method. We can also include all the side effects in this block, including that a  plate type is changed and fees are collected. That would be the integration test. Then when you test those methods by themselves in their own it blocks, those would be the unit tests.